### ISSUE TITLE
Time changed

### ISSUE THREAD

**Issue by syneinsindia** [saifulcoder/adms-server-ZKTeco#14]
After connecting with the application and device time changes to +2:30 hrs

Are there any commands to change the time

1. Are the any options to transfer one device's data to another device like ADMS V3.1-165 (iclock Server)

**Comment by mdestafadilah** [comment]
whats your hardware using?

**Comment by singok** [comment]
I am using ZKT ECO

**Comment by mdestafadilah** [comment]
can you give version? i have not thats device zkteco

**Comment by veloxkaran** [comment]
I have the issues with time changed . same as singok :: zkteco k40 pro

**Comment by mdestafadilah** [comment]
> I have the issues with time changed . same as singok :: zkteco k40 pro

do you have time in weekend? i can give my time to help this issue. i need you attach my ip public to your fingerprint harware.

**Comment by veloxkaran** [comment]
Sure. Lets connect : karankhanal775@gmail.com. Feel free to mail your available time  at weekend any time 
.

**Comment by mdestafadilah** [comment]
> Sure. Lets connect : [karankhanal775@gmail.com](mailto:karankhanal775@gmail.com). Feel free to mail your available time at weekend any time .

oke, i will contacted you.

**Comment by Myabadhon** [comment]
> After connecting with the application and device time changes to +2:30 hrs
> 
> Are there any commands to change the time
> 
> 1. Are the any options to transfer one device's data to another device like ADMS V3.1-165 (iclock Server)

I'm facing the same issue when configuring ADMS. The device time changed to +2 hour.

**Comment by veloxkaran** [comment]
Issue Solved! time zone ( UTC+ in min ). This solved my problems .

**Comment by saifulcoder** [comment]
You can set the timezone when the device will perform a handshake.

[iclockController](https://github.com/saifulcoder/adms-server-ZKTeco/tree/main/app/Http/Controllers/iclockController.php) line : 46


![Image](https://github.com/user-attachments/assets/eae9fab7-3ed7-41fd-8dde-855457fdd87d)

**Comment by veloxkaran** [comment]
Yes! I uncommented the timezone where the value is set (in a minute ),
which solved my problem.

**Comment by Myabadhon** [comment]
@saifulcoder @veloxkaran Thanks a lot. It's working.

**Comment by sanketgroup** [comment]
i removed comment from
"TimeZone=7\r\n"
(I think this is not in minute, it is in hours)

problem is how can i define 5.5 instead of 7? (+5.5 is IST)

I tried 5.5 but it consider as 5 in actual display.

**Comment by JanithaU** [comment]
Thank you @veloxkaran , you saved my day

**Comment by veloxkaran** [comment]
Hi Everyone.  We have done well and are working fine in zkteco device. Do
you have any solutions for the Hikvision biometric device?

**Comment by mdestafadilah** [comment]
> Hi Everyone.  We have done well and are working fine in zkteco device. Do
> you have any solutions for the Hikvision biometric device?
> […](#)

can you share details< like type number, year product, and image of product. thanks

**Comment by veloxkaran** [comment]
Model Number: DS-K1A8503EF-B
Hikvision 2024 product

**Comment by rangasl** [comment]
> i removed comment from "TimeZone=7\r\n" (I think this is not in minute, it is in hours)
> 
> problem is how can i define 5.5 instead of 7? (+5.5 is IST)
> 
> I tried 5.5 but it consider as 5 in actual display.

anyone solved this?

**Comment by Amloner** [comment]
Use "TimeZone=330\r\n"
It worked for me in ZK40 Pro

**Comment by mdestafadilah** [comment]
> ZK40 Pro

same as here >> https://www.tokopedia.com/kiossolusi/mesin-absensi-fingerprint-zkteco-k40-pro-sidik-jari-battery, bro?

**Comment by PriyaViewzen** [comment]
After connecting with the nodejs and biometric essL X990 time changes to +2:30 hrs.
How to fix this issue. Anybody help me out from this.

**Comment by Myabadhon** [comment]
If you're using a domain name in the ADMS configuration, this issue may
occur occasionally. Try using the server's IP address instead of the domain
name. That resolved the issue on my end.

**Comment by osmanahmadxai** [comment]
Hi, 

I decompiled ZKTeco's own ZKBio Time 9.0.6 to confirm how their server handles this.

Setting the time: the device takes its clock from its ADMS master, so a wrong offset means the server pushed the wrong local time. Answer the command poll (GET /iclock/getrequest?SN=...) with:


C:1:SET OPTIONS DateTime=2026-07-30 13:45:00
It's a plain YYYY-MM-DD HH:MM:SS string of your local time — not the integer form, not a TimeZone= field. ZKBioTime just pushes its own datetime.now() on handshake.

Transferring data between devices: no direct clone — the server reads from A, writes to B:


C:1:DATA QUERY tablename=user,fielddesc=*,filter=*         # users
C:1:DATA QUERY tablename=templatev10,fielddesc=*,filter=*  # fingerprints
C:1:DATA QUERY tablename=facev7,fielddesc=*,filter=*       # faces
The device uploads the rows, then write them to the target with DATA UPDATE table ...
